ecs-cli/modules/utils/compose/convert_task_definition.go

```diff
@@ -104,7 +104,27 @@ func ConvertToTaskDefinition(taskDefinitionName string, context *project.Context
 
 	// Add named volume configs:
 	if context.Project.VolumeConfigs != nil {
-		for name := range context.Project.VolumeConfigs {
+		for name, config := range context.Project.VolumeConfigs {
+			if config != nil {
+				if config.Driver != "" {
+					log.WithFields(log.Fields{
+						"volume name": name,
+						"option name": "driver",
+					}).Warn("Skipping unsupported YAML option...")
+				}
+				if len(config.DriverOpts) != 0 {
+					log.WithFields(log.Fields{
+						"volume name": name,
+						"option name": "driver_opts",
+					}).Warn("Skipping unsupported YAML option...")
+				}
+				if config.External.External {
```

Comment:
I'm not sure I completely sure I understand the external field, but to me it looks like you might need to do more of a check here. See these:

https://github.com/docker/libcompose/blob/master/yaml/external.go#L4
https://docs.docker.com/compose/compose-file/compose-file-v2/#external

Reply:
Actually never mind, I think your code works.

Reply:
I tested that the warnings work properly. 👍

Reply:
If someone else is wondering why this works:
The `UnmarshalYAML` method for `yaml.External` properties always sets the `boolean` to true:
https://github.com/docker/libcompose/blob/master/yaml/external.go#L34
So if the `VolumeConfig.External` property is set (either directly to `true` or by setting a `name`), it's own `External` `boolean` property will always be `true`.